app/controllers/groups_controller.rb

```diff
@@ -164,35 +164,34 @@ def join
   end
 
   def leave
-    if params[:memberid].blank?
-      memberid = current_user.id
-    else
-      memberid = params[:memberid]
-      membername = User.where(id: memberid).first.name
-    end
-
-    group_name = Group.where(id: params[:groupid]).first.name
+    member_id = params[:memberid] || current_user.id
+    group_member = GroupMember.find_by(userid: member_id, groupid: params[:groupid])
 
     # Cannot leave When you are the only leader
-    is_leader = GroupMember.where(userid: memberid, groupid: params[:groupid], leader: true).count
-    are_leaders = GroupMember.where(groupid: params[:groupid], leader: true).count
-    if is_leader == 1 && are_leaders == is_leader
+    if group_member.group.leader_ids == [member_id]
```

Comment:
Also, I know we only use `group_member.group` twice in this method, but it might make the flow a little clearer if we assign `group` to a variable and operate on it directly instead of going through the `group_member.group`.